Remove unnecessary trailing semicolons

Andrew-Fryer opened the pull request:
#### b4dbf7a9e46007ef1db2d1ae69cc3b428ef2dd25
<pre>
Remove unnecessary trailing semicolons
<a href="https://bugs.webkit.org/show_bug.cgi?id=247694">https://bugs.webkit.org/show_bug.cgi?id=247694</a>
<a href="https://rdar.apple.com/102434342">rdar://102434342</a>

Reviewed by Tim Nguyen.

remove unecessary semicolons after closing brace

* Source/WebCore/css/SelectorChecker.cpp:
(WebCore::SelectorChecker::matchRecursively const):
* Source/WebCore/platform/ScrollView.h:
(WebCore::ScrollView::setVerticalScrollbarMode):

Canonical link: <a href="https://commits.webkit.org/280878@main">https://commits.webkit.org/280878@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/03113123ef4ab8761fbc9efaca97696f3d4d8d6f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/57752 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/37080 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/10228 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/61374 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/8197 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/59880 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/44716 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/8385 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/46793 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/5815 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/59782 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/34775 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/49905 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/27618 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/31545 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/7198 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/7201 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/53497 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/7468 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/63056 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/1666 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/7542 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/54011 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/1672 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/49916 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/54129 "Found 3 new API test failures: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/memory-pressure, /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/text/state-changed, /WebKitGTK/TestInspectorServer:/webkit/WebKitWebInspectorServer/http-test-page-list (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12807 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/1411 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/32909 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/33995 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/35079 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/33740 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Andrew-Fryer commented on Source/JavaScriptCore/parser/Parser.cpp:
```diff
@@ -1447,7 +1447,7 @@ template <class TreeBuilder> TreeStatement Parser<LexerType>::parseForStatement(
             return lexicalVariables;
         }
         return { };
-    };
```
I think we should keep this semicolon because it is an assignment.
See https://webkit.org/code-style-guidelines/#other-punctuation, which has "auto a = [] { };"

Andrew-Fryer commented on Source/JavaScriptCore/parser/Parser.cpp:
```diff
@@ -2687,7 +2687,7 @@ template <class TreeBuilder> bool Parser<LexerType>::parseFunctionInfo(TreeBuild
 
     auto performParsingFunctionBody = [&] {
         return parseFunctionBody(context, syntaxChecker, startLocation, startColumn, functionStart, functionNameStart, parametersStart, constructorKind, expectedSuperBinding, functionBodyType, functionInfo.parameterCount);
-    };
```
I think we should keep this semicolon because it is an assignment.
See https://webkit.org/code-style-guidelines/#other-punctuation, which has "auto a = [] { };"

Andrew-Fryer commented on Source/WebCore/workers/WorkerMessagingProxy.cpp:
```diff
@@ -151,7 +151,7 @@ void WorkerMessagingProxy::startWorkerGlobalScope(const URL& scriptURL, PAL::Ses
         initializationData.clientIdentifier.value_or(ScriptExecutionContextIdentifier { }),
         m_scriptExecutionContext->advancedPrivacyProtections(),
         m_scriptExecutionContext->noiseInjectionHashSalt()
-    };
```
I think we should keep this semicolon because it is a declaration.

Andrew-Fryer commented on Source/WebCore/workers/service/context/ServiceWorkerThread.cpp:
```diff
@@ -106,7 +106,7 @@ static WorkerParameters generateWorkerParameters(const ServiceWorkerContextData&
         { },
         advancedPrivacyProtections,
         noiseInjectionHashSalt
-    };
```
I think we should keep this semicolon because it is a statement.

Andrew-Fryer commented on Source/WebCore/workers/shared/context/SharedWorkerThreadProxy.h:
```diff
@@ -96,6 +96,6 @@ class SharedWorkerThreadProxy final : public ThreadSafeRefCounted<SharedWorkerTh
     RefPtr<CacheStorageConnection> m_cacheStorageConnection;
     bool m_isTerminatingOrTerminated { false };
     ClientOrigin m_clientOrigin;
-};
```
I think we should keep this semicolon because it is ending a class.

Andrew-Fryer commented on Source/JavaScriptCore/parser/Parser.cpp:
```diff
@@ -2426,7 +2426,7 @@ template <class TreeBuilder> typename TreeBuilder::FormalParameterList Parser<Le
         auto binding = context.createBindingLocation(location, name, position, position, AssignmentContext::DeclarationStatement);
         context.appendParameter(parameters, binding, 0);
         ++parameterCount;
-    };
```
I think we should keep this semicolon because it is an assignment.
See https://webkit.org/code-style-guidelines/#other-punctuation, which has "auto a = [] { };"

vitorroriz reviewed (changes requested):
Please remove the 'Pull Request Template' and below from the commit message.

Andrew-Fryer committed:
Remove unnecessary trailing semicolons
https://bugs.webkit.org/show_bug.cgi?id=247694
rdar://102434342

Reviewed by Tim Nguyen.

remove unecessary semicolons after closing brace

* Source/WebCore/css/SelectorChecker.cpp:
(WebCore::SelectorChecker::matchRecursively const):
* Source/WebCore/platform/ScrollView.h:
(WebCore::ScrollView::setVerticalScrollbarMode):

Canonical link: https://commits.webkit.org/280878@main
